src/PriceCalculatorKata/IProduct.cs

```diff
@@ -0,0 +1,11 @@
+namespace PriceCalculatorKata;
+public interface IProduct
+{
+    public static float Tax { get; set; }
+    public static float Discount { get; set; }
+    public string? Name { get; set; }
+    public int UPC { get; set; }
+    public double Price { get; set; }
+    public string ReportAboutPrice();
```

Comment:
I think you need just one model for product, and separate classes with these methods

Reply:
like separate services/managers to deal with the reports, taxes, discounts,...

Reply:
so this interface is not really required for Product DTO. Just move out these methods to separate classes that will specialized for certain logic.